v3/src/components/case-tile-common/inspector-panel/case-tile-inspector.tsx

```diff
@@ -36,15 +39,55 @@ export const CaseTileInspector = ({ tile, show, showResizeColumnsButton }: IProp
       case "datasetInfo":
         setShowInfoModal(true)
         break
-      case "resizeColumns":
-        //TODO move log to respective handler
-        caseTileModel?.applyModelChange(() => {}, {
-          log: logStringifiedObjectMessage("resizeColumns: %@", {dataContext: data?.name}, "table")
-        })
-        break
     }
   }
 
+  const resizeAllColumns = () => {
+    data?.collections.forEach((collection) => {
+      collection.attributes.forEach((attr) => {
+        if (attr) {
+          const attrId = attr?.id
+          const longestContentWidth = findLongestContentWidth(attr)
+          tableModel?.applyModelChange(() => {
+            tableModel?.setColumnWidth(attrId, longestContentWidth)
+          }, {
+            log: {message: "Resize all columns", args:{}, category: "table"},
+            undoStringKey: "DG.Undo.caseTable.resizeColumns",
+            redoStringKey: "DG.Redo.caseTable.resizeColumns"
+          })
+        }
+      })
+    })
+  }
+
+  const findLongestContentWidth = (attr: IAttribute) => {
+    let longestWidth = 0
+    const widthAdjustment = 25 // width calculation is too wide without this adjustment
```

Comment:
Do you know why the calculation is off? Maybe it could be fixed by matching the font or other styling? If doing something as complex as creating and then destroying a hidden span is giving you an inaccurate width, maybe it would be better to approximate it in a simpler way, like `string.length * 8`.